src/flint/fmpq_abs_series.jl

```diff
@@ -925,16 +925,18 @@ end
 #
 ###############################################################################
 
-function PowerSeriesRing(R::FlintRationalField, prec::Int, s::AbstractString; model=:capped_relative, cached = true)
-   S = Symbol(s)
-
+function PowerSeriesRing(R::FlintRationalField, prec::Int, s::Symbol; model=:capped_relative, cached = true)
    if model == :capped_relative
-      parent_obj = FmpqRelSeriesRing(prec, S, cached)
+      parent_obj = FmpqRelSeriesRing(prec, s, cached)
    elseif model == :capped_absolute
-      parent_obj = FmpqAbsSeriesRing(prec, S, cached)
+      parent_obj = FmpqAbsSeriesRing(prec, s, cached)
    else
       error("Unknown model")
    end
 
    return parent_obj, gen(parent_obj)
 end
+
+function PowerSeriesRing(R::FlintRationalField, prec::Int, s::AbstractString; model=:capped_relative, cached = true)
```

Comment:
That's a lot of boilerplate to duplicate every method for `String/Symbol`. I guess you also considered this alternative, but what about just having a type alias like `const VarLike = Union{AbstractString, Symbol}` and using it in this type signature, with a call to `Symbol(s)` when calling out to the lower-level constructor?

Reply:
Yeah I could do this, but as you know, I'm not massively in favour of too much metaprogramming, due to the obfuscation it can cause.